build.zig

```diff
@@ -4,55 +4,55 @@ const Builder = @import("std").Build;
 
 pub const BuildContext = struct {
     builder: *Builder,
-    build_options: *Step.Options = undefined,
-
-    // Steps
-    kernel: *Step.Compile = undefined,
-    grub: *Step.Run = undefined,
-    qemu: *Step.Run = undefined,
-    run: *Step = undefined,
-
-    // Installed files
-    install_iso_folder: *Step.InstallDir = undefined,
-    install_disk_image: *Step.InstallFile = undefined,
-    install_kernel: *Step.InstallArtifact = undefined,
-
-    // options
     optimize: std.builtin.Mode = undefined,
     iso_source_dir: []const u8 = undefined,
     name: []const u8 = undefined,
+    bootloader: enum { grub, limine } = .grub,
     posix: bool = false,
     ci: bool = false,
 };
 
 pub fn build(b: *Builder) !void {
     var context = BuildContext{ .builder = b };
 
-    context.iso_source_dir = context.builder.option(
+    context.iso_source_dir = b.option(
         []const u8,
         "iso_dir",
         "Specify the iso directory source",
     ) orelse "iso";
 
-    context.name = context.builder.option(
+    context.name = b.option(
         []const u8,
         "name",
         "Specify a name for output binary",
     ) orelse "kfs.elf";
 
-    context.posix = context.builder.option(
+    context.posix = b.option(
         bool,
         "posix",
         "Enable this flag if strict POSIX conformance is wanted",
     ) orelse false;
 
+    context.bootloader = b.option(
+        @TypeOf(context.bootloader),
+        "bootloader",
+        "Specify the bootloader to use",
+    ) orelse .grub;
+
     context.ci = b.option(bool, "ci", "Build the kernel for CI") orelse false;
 
-    context.optimize = context.builder.standardOptimizeOption(.{});
+    context.optimize = b.standardOptimizeOption(.{});
+
+    // Build steps
+    const kernel = @import("build/kernel.zig").build_executable(&context);
+    const themes = @import("build/themes.zig").install_themes(&context);
+    const install_disk_image = @import("build/iso.zig").build_iso(&context, kernel);
+    const uninstall_themes = @import("build/themes.zig").uninstall_themes(&context);
+    _ = @import("build/qemu.zig").add_step_run(&context, install_disk_image);
 
-    @import("build/disk_image.zig").install_iso_folder(&context);
-    @import("build/kernel.zig").build_executable(&context);
-    @import("build/themes.zig").install_themes(&context);
-    @import("build/disk_image.zig").build_disk_image(&context);
-    @import("build/qemu.zig").add_step_run(&context);
+    // Dependencies
+    kernel.step.dependOn(&themes.step);
+    b.getInstallStep().dependOn(&install_disk_image.step);
+    b.getInstallStep().dependOn(&install_disk_image.step);
```

Comment:
seems like a mistake (I'm not sure if you wanted another dependency or if you just accidentally duplicated the line)

Reply:
Done